Makefile.core.mk

```diff
@@ -385,8 +385,10 @@ copy-templates:
 			sed "1s|^|$${warning}\n\n|" $$profile > manifests/charts/$$chart/files/profile-$$(basename $$profile) ; \
 		done; \
 		[[ "$$chart" == "ztunnel" ]] && flatten="true" || flatten="false" ; \
+		[[ "$$chart" == "istio-control/istio-discovery" ]] && componentFlatten="pilot" || componentFlatten="" ; \
```

Comment:
Rather than do this here, can we not just put a localized `xxx_profile.yaml` in each subcomponent that needs this?


So everybody gets `zzz_profile.yaml` which is copied out centrally.

And only the charts that need fixes get `xxx_values-scope-compat.yaml`

That reduces the sprawl of this fix a bit and relies less on `sed`/Makefile.

(that might require twiddling the prefix of the `_profile.yamls` to get the precedence order to come out, but we should be able to do that)

Reply:
It could, the only issue really is that it happens to only work since we do this last. The ztunnel one, for example, is intermixed, so then we probably still need the hacks there?

Reply:
I might be wrong but

```
pilot/
  - yyy_profile.yaml (same-for-all)
  - zzz_descope-legacy.yaml (chart-local)
```

```
ztunnel/
  - yyy_profile.yaml
```

Should work?

Reply:
Yep that should, I just meant the ztunnel hack (FLATTEN_GLOBALS_REPLACEMENT) needs to persist

Reply:
Yep for now it can.

If we actually do templating for stuff like image values and such as per https://github.com/istio/istio/issues/52034 then I suspect we can just move _all_ globals handling into shared templates - none in individual charts themselves. That way we can prevent uncontrolled use of globals between charts, and avoid having `if values.global.foo || values.foo` in any chart. If we do that we can also likely remove `FLATTEN_GLOBALS_REPLACEMENT` as a unique ZT problem here as well.

But - yeah. For now, for `FLATTEN_COMPONENT_REPLACEMENT`, I'd rather trend in the direction of per-chart extra override yamls, rather than central makefile stuff.

Reply:
done